### PR TITLE
Updating Dev testing env

### PR DESCRIPTION
**Pull request by didierofrivia**
This PR updates our development environment for testing and the e2e test for the remote address using CEL expressions and predicates.

### CODE REVIEW

File: utils/deploy/envoy.yaml [hunk 1]
```diff
@@ -337,7 +337,7 @@ data:
                                           "data": [
                                             {
                                               "selector": {
-                                                "selector": "filter_state.wasm\\.kuadrant\\.identity\\.userid",
+                                                "selector": "auth.identity.userid",
```
Comment (alexsnaps) [inline utils/deploy/envoy.yaml:340]: This is probably not something one would ever want to do. 
All `selector` usages should be avoided. This should be an `expression` here.

File: utils/deploy/envoy.yaml [hunk 2]
```diff
@@ -212,7 +204,7 @@ data:
                                           "scope": "rlp-ns-B/rlp-name-B",
                                           "data": [
                                             {
-                                              "static": {
+                                              "expression": {
                                                 "key": "rlp-ns-B/rlp-name-B/limit-not-to-be-activated",
                                                 "value": "1"
```
Comment (alexsnaps) [inline utils/deploy/envoy.yaml:209]: While this would work, this also changed the type tho! It's now `int(1)` ... but will be erased and will be `1`, just as if you had it as `"\"1\""` in there, which would be closer semantically to what this was before...

tl;dr _not wrong_, but... ;)

**Review by eguzki — APPROVED** [review]
